GMX-Bots/Large-Open-and-Increase-Positions/src/agent.spec.ts

```diff
@@ -0,0 +1,542 @@
+import { Finding, ethers, FindingSeverity, FindingType, HandleTransaction, TransactionEvent } from "forta-agent";
+import { createAddress, TestTransactionEvent } from "forta-agent-tools/lib/tests";
+import { provideTransactionHandler } from "./agent";
+import NetworkData from "./network";
+import { VAULT_CONSTANTS } from "./constants";
+import { ethersBnToBn } from "./utils";
+
+const { INCREASE_POSITION_EVENT, UPDATE_POSITION_EVENT, THRESHOLD } = VAULT_CONSTANTS;
+const TEST_PRICE_MULTIPLIER = 30;
+const MOCK_OTHER_EVENT: string = "event UpdateFundingRate(address token, uint256 fundingRate)";
+const MOCK_EVENT_ABI: string[] = [INCREASE_POSITION_EVENT, UPDATE_POSITION_EVENT];
+
+type mockConstantsType = {
+  MOCK_VAULT_ADDRESS: string;
+  MOCK_OTHER_CONTRACT: string;
+  MOCK_ACCOUNT: string;
+  MOCK_TOKEN: string;
+  MOCK_IFACE: ethers.utils.Interface;
+};
+
+const MOCK_CONSTANTS: mockConstantsType = {
+  MOCK_VAULT_ADDRESS: createAddress("0xa1"),
+  MOCK_OTHER_CONTRACT: createAddress("0x99"),
+  MOCK_ACCOUNT: createAddress("0x11a"),
+  MOCK_TOKEN: createAddress("0x85e"),
+  MOCK_IFACE: new ethers.utils.Interface(MOCK_EVENT_ABI),
+};
+
+const { MOCK_VAULT_ADDRESS, MOCK_OTHER_CONTRACT, MOCK_IFACE, MOCK_ACCOUNT, MOCK_TOKEN } = MOCK_CONSTANTS;
+
+const eventParams = {
+  updatePositionKey: ethers.utils.formatBytes32String("key1"),
+  increasePositionKey: ethers.utils.formatBytes32String("key2"),
+  largeSizeDelta: ethers.BigNumber.from(50000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_MULTIPLIER - 3)),
+  smallSizeDelta: ethers.BigNumber.from(100).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_MULTIPLIER)),
+  largePositionSize: ethers.BigNumber.from(50000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_MULTIPLIER - 3)),
+  largePositionSize2: ethers.BigNumber.from(600000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_MULTIPLIER - 3)),
+  smallPositionSize: ethers.BigNumber.from(100).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_MULTIPLIER)),
+};
+
+const {
+  updatePositionKey,
+  increasePositionKey,
+  largeSizeDelta,
+  smallSizeDelta,
+  smallPositionSize,
+  largePositionSize,
+  largePositionSize2,
+} = eventParams;
```

Comment:
why do you like to do this all the time, I have this question in mind since your first PR. why not:
```suggestion
const updatePositionKey: string = ethers.utils.formatBytes32String("key1");
const increasePositionKey:  string = ethers.utils.formatBytes32String("key2");
const largeSizeDelta: ethers.BigNumber = ethers.BigNumber.from(50000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_MULTIPLIER - 3)),
...
```